lib/Parser/parse.cpp

```diff
@@ -4427,6 +4438,17 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, ParseNodePtr pnodeFncPare
                 // Start the var list.
                 pnodeFnc->sxFnc.pnodeVars = nullptr;
                 m_ppnodeVar = &pnodeFnc->sxFnc.pnodeVars;
+
+                // We can't merge the param scope and body scope any more as the nested methods may be capturing params.
+                Scope* paramScope = pnodeFnc->sxFnc.pnodeScopes->sxBlock.scope;
+                if (!paramScope->GetCanMergeWithBodyScope())
+                {
+                    Assert(!pnodeFnc->sxFnc.IsSimpleParameterList());
+                    // Now add a new symbol reference for each formal in the param scope to the body scope.
+                    paramScope->ForEachSymbol([this](Symbol* param) {
+                        this->CreateVarDeclNode(param->GetPid(), param->GetSymbolType(), false, nullptr, false);
```

Comment:
Shouldn't the second param be STVariable instead of STFormal? I believe the param->GetSymbolType will return STFormal.